src/test/run-pass/union/union-pat-refutability.rs

```diff
@@ -29,7 +29,7 @@ fn is_zero(v: Value) -> bool {
     unsafe {
         match v {
             Value { tag: Tag::I, u: U { i: 0 } } => true,
-            Value { tag: Tag::F, u: U { f: 0.0 } } => true,
+            Value { tag: Tag::F, u: U { u: 0 } } => true,
```

Comment:
It is annotated with unsafe, but unsafe doesn't allow everything, so IMO its okay to disallow it. Especially as making unsafe feel more comfortable to write than non unsafe code is not really a priority of Rust.